assets/js/phoenix.js

```diff
@@ -376,11 +376,11 @@ export class Channel {
     this.rejoinTimer = new Timer(() => {
       if(this.socket.isConnected()){ this.rejoin() }
     }, this.socket.rejoinAfterMs)
-    this.socket.onError(() => this.rejoinTimer.reset())
+    this.socket.onError(() => this.rejoinTimer.reset(), topic)
```

Comment:
to avoid clashes, let's keep the on[..] callbacks accepting a single function argument, and we will always generate and return the unique reference. This will ensure no reference clashes, and `off` will accept a list of refs to drop. This will require the channel to track its state change refs, but that will be pretty simple:

```diff
class Channel {
  constructor(...){
    ...
+   this.stateChangeRefs = []
  }
  ...
-  this.socket.onOpen(...)
+  this.stateChangeRefs.push(this.socket.onOpen(...))
-  this.socket.onClose(...)
+  this.stateChangeRefs.push(this.socket.oClose(...))